src/components/markdown/markdown-parser.ts

```diff
@@ -37,6 +38,17 @@ export async function markdownToHTML(
     const file = await unified()
         .use(remarkParse)
         .use(remarkGfm)
+        .use(() => {
+            return (tree: Node) => {
+                // Run the parse5.serialize function on all html elements,
+                // to minify it to get rid of extra whitespace that causes rendering issues
+                // when elements are opened/closed with empty lines between them.
+                visit(tree, 'html', (node: any) => {
+                    const documentFragment = parse5.parseFragment(node.value);
+                    node.value = parse5.serialize(documentFragment);
```

Comment:
This doesn't quite handle html properly.

Look at the "HTML" example on this branch vs latest release for example:

**This PR:**
<img width="590" alt="image" src="https://github.com/user-attachments/assets/a37e2ec3-e70e-4a1c-b334-66e4e37ce540">

**Main:**
<img width="590" alt="image" src="https://github.com/user-attachments/assets/6c80d06f-e41d-47bd-885e-dc97d32f4fbb">

Also, the inline `<code>` element in the "Code" example isn't handled properly:

**This PR:**
<img width="590" alt="image" src="https://github.com/user-attachments/assets/daa931a3-1332-4ae5-97d6-260ec08e87db">

**Main:**
<img width="590" alt="image" src="https://github.com/user-attachments/assets/615a481f-b286-4752-9fee-9ae6aad79a4d">

Reply:
@john-traas is this something you have time to take a look at?

Reply:
I've been out sick for a few days but will be back from tomorrow. I'll look further into this and try and figure out what's going on. 
As I mentions when I first suggested something like this, I hadn't tested more use cases and it was just an initial exploration into how we might fix it.

Reply:
On looking further into this I think we might need to delegate responsibility to the consumer.

Reply:
So, @TommyLindh2, either the consumer using the `limel-markdown` component needs to pre-process the data before sending it to `limel-markdown`, or, and I would say preferably, the component responsible for storing the data in the first place should process it and make sure the data isn't broken.

Reply:
Yeah, I think it makes a lot of sense as well.
In this case we'll try to sanitize the content before passing it to limel-markdown.